src/connection-manager/connections.ts

```diff
@@ -55,11 +62,35 @@ async function handleWsMessageTypes(
     if (ledger_hashes.includes(validationData.ledger_hash)) {
       validationData.networks = networks
     }
-    void agreement.handleValidation(data as ValidationRaw)
+
+    // Get network of the validation if ledger_hash is not in cache.
+    const validationNetworkDb: DatabaseValidator | undefined = await query(
+      'validators',
+    )
+      .select('*')
+      .where('signing_key', validationData.validation_public_key)
+      .first()
+    const validationNetwork =
+      validationNetworkDb?.networks ?? validationData.networks
```

Comment:
nit: I think the code would be a bit easier to read if this was in a helper function.